src/main/java/br/ufsc/bridge/soap/http/SoapMessageBuilder.java

```diff
@@ -94,9 +94,9 @@ protected SOAPMessage soapMessage(Object data) throws SoapCreateMessageException
 	}
 
 	protected Document jaxbObjectToDocument(Object data) throws JAXBException, ParserConfigurationException {
-		JAXBContext jc = JAXBContext.newInstance(data.getClass());
+		JAXBContext jc = JAXBContextWrapper.newInstance(data.getClass());
 
-		Document document = DocumentBuilderFactory.newInstance().newDocumentBuilder().newDocument();
+		Document document = DOCUMENT_BUILDER_FACTORY.newDocumentBuilder().newDocument();
```

Comment:
Segundo a oracle Document builder factory não é thread safe
https://docs.oracle.com/cd/E17802_01/webservices/webservices/docs/1.5/api/javax/xml/parsers/DocumentBuilderFactory.html
Nesse tópico é discutido a implementação de um pool de document builder com auxilio ConcurrentLinkedQueue, talvez ajude colocar uma LinkedBlockingQueue ajude mais, assim se não tiver nada no pool vai ficar esperando e no final do processo recoloca o builder no pool de volta
https://community.oracle.com/thread/1626108

Reply:
Acho que vc fala sobre o `DocumentBuilderFactory`, eu li errado em algum lugar que era, vou reverter isso.
Sobre a estratégia do pool, eles só recomendam se for fazer parse de muitos documentos por segundo pelo que li.